server/datastore/mysql/hosts.go

```diff
@@ -510,6 +511,12 @@ func (ds *Datastore) applyHostFilters(opt fleet.HostListOptions, sql string, fil
 		operatingSystemJoin = `JOIN host_operating_system hos ON h.id = hos.host_id`
 	}
 
+	munkiFilter := "TRUE"
+	if opt.MunkiIssueIDFilter != nil {
+		munkiFilter = "EXISTS (SELECT 1 FROM host_munki_issues hmi WHERE hmi.host_id = h.id AND hmi.munki_issue_id = ?)"
```

Comment:
Maybe we can consider this a nit, but this pattern was not recommended to be used for performance reasons. I wonder if we can change it for a JOIN without too much effort?

We can also think about this and change all of them at once, that's why this _could_ be considered a nit. Up to you.

Reply:
Gotcha, yeah I think it could be converted to a JOIN without too much effort. Do you have a reference to that performance article for my own benefit? I'd be interested to read about the details and comparison.

Reply:
We should sift through the slides. The online articles are varied, couldn't find anything on percona itself, but the consensus seems to be the same.

Reply:
Ok, in any case I updated the condition to a JOIN with a WHERE.